pkg/controller/data_upload_controller.go

```diff
@@ -292,19 +292,20 @@ func (r *DataUploadReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	} else if du.Status.Phase == velerov2alpha1api.DataUploadPhaseInProgress {
 		log.Info("Data upload is in progress")
 		if du.Spec.Cancel {
-			fsBackup := r.dataPathMgr.GetAsyncBR(du.Name)
-			if fsBackup == nil {
-				return ctrl.Result{}, nil
-			}
 			log.Info("Data upload is being canceled")
 
-			// Update status to Canceling.
+			// Update status to Canceled
 			original := du.DeepCopy()
-			du.Status.Phase = velerov2alpha1api.DataUploadPhaseCanceling
+			du.Status.Phase = velerov2alpha1api.DataUploadPhaseCanceled
```

Comment:
For case 2, instead of patching the DUCR directly here, let's call `OnDataUploadCancelled` instead, it does something more:

- Update the completionTimeStamp
- Cleanup intermediate object

Reply:
@shubham-pampattiwar I think the updated version still doesn't handle case 2.
When there is no fsbackup, then we set Canceled directly, but when there is, we need to cancel the fsbackup rather than just setting the phase.

Reply:
@shubham-pampattiwar 
For 1, we need to call `fsBackup.Cancel()` and not set `du.Status.Phase` (the same as the original code).
For 2, we just need to call `OnDataUploadCancelled`